File: Leetcode/weekly contest - 277/C.py
class Solution:
    def findLonely(self, nums: List[int]) -> List[int]:
        map = {}
        for item in nums:
            if map.get(item):
                map[item] += 1
            else :
                map[item] = 1

        result = []
        for item in map:
            if map[item] == 1 :
                if map.get(item - 1) or map.get(item + 1) :
                    continue
                else :
                    result.append(item)

        return result

File: Leetcode/weekly contest - 277/C.js
var findLonely = function (nums) {
  if (!nums) return [];
  let charMap = {};
  for (let char of nums) {
    if (charMap[char]) {
      charMap[char]++;
    } else {
      charMap[char] = 1;
    }
  }
  let ans = [];
  for (let item of nums) {
    let isTrue = !charMap[item - 1] && !charMap[item + 1] && charMap[item] == 1;
    if (isTrue) {
      ans.push(item);
    }
  }

  return ans;
};
